### ISSUE TITLE
Lighting glitch since "Variable name fix + structure creation unrolling in lighting code"

### ISSUE THREAD

**Issue by mtango688** [minetest/minetest#7118]
##### Issue type

 - Bug report
 
##### Minetest version

```
See the screenshots (bcdb3d5)
```

##### OS / Hardware

Operating system:
CPU: AMD

GPU model: nVidia 1050
OpenGL version:  ?

##### Summary

When using commit bcdb3d5 or newer, changes to mesecon (powerred/unpowered) causes other lights nearby to lose their spread.

##### Steps to reproduce

Create a new world with Minetest_game and add Mesecons as a mod.
Change to night and place torches and mesecons.
Change the state of the mesecons.
![before](https://user-images.githubusercontent.com/18478751/37237698-17b0dc40-23e5-11e8-8641-09a9debe5f2c.jpg)
![after](https://user-images.githubusercontent.com/18478751/37237700-1a0d5f36-23e5-11e8-8c5a-568824ba4ab6.jpg)

**Comment by rubenwardy** [comment]
May be this line: https://github.com/minetest/minetest/commit/bcdb3d56c0c3e19ebe1437f345b425fdc20f3926#diff-d8d3b69601fbe8b3c44441473970f0a0R1117

Try changing to

```lua
const ContentFeatures &newf = ndef->get(newnode);
```

**Comment by mtango688** [comment]
I don't think so. It's probably better if you can observe the bug first-hand.

**Comment by rubenwardy** [comment]
Can confirm that bcdb3d5 is the first bad commit.

**Comment by nOOb3167** [comment]
typo, Y instead of Z

> MapNode node = block->getNodeNoCheck(relpos.X, relpos.Y, relpos.Y, &is_valid);

> MapNode node = block->getNodeNoCheck(relpos.X, relpos.Y, relpos.Z, &is_valid);

seems to resolve the issue

unavoidable considering all the XYZ soup - did not notice until the 3rd readthrough.

**Comment by SmallJoker** [comment]
@nOOb3167 Thanks for your research. This issue is now fixed: ee20433

**Comment by nerzhul** [comment]
Ty @smalljoker. It seems a missing force push didn't pass on my pr before merge, as i replaced (x,y,z) with v3s16 for that call...
